modules/virt-configuring-vm-real-time.adoc

```diff
@@ -49,56 +49,127 @@ spec:
+.. Disable the `irqbalance` service by using the following command:
 +
 [source,terminal]
 ----
-$ grubby --update-kernel=ALL --args="default_hugepagesz=1GB hugepagesz=1G hugepages=8"
+$ systemctl disable irqbalance && systemctl stop irqbalance
 ----
-
-.. To achieve low-latency tuning by using the `cpu-partitioning` profile in the TuneD application, run the following commands:
+.. Install the necessary packages by running the following command:
 +
 [source,terminal]
 ----
-$ dnf install -y tuned-profiles-cpu-partitioning
+$ dnf install kernel-rt tuned cloud-init realtime-tests
 ----
+.. To achieve low-latency tuning by using the `realtime-virtual-guest` profile in the TuneD application, run the following commands:
 +
 [source,terminal]
 ----
-$ echo isolated_cores=2-9 > /etc/tuned/cpu-partitioning-variables.conf
+$ dnf --enablerepo=rt install -y tuned-profiles-realtime
 ----
-The first two CPUs (0 and 1) are set aside for house keeping tasks and the rest are isolated for the real-time application.
 +
 [source,terminal]
 ----
-$ tuned-adm profile cpu-partitioning
+$ dnf --enablerepo=nfv install -y tuned-profiles-nfv-guest
 ----
+.. Edit `/etc/tuned/realtime-virtual-guest-variables.conf`:
++
+[source,conf]
+----
+#
+# Variable settings below override the definitions from the
+# /etc/tuned/realtime-variables.conf file.
+
+#
+# Core isolation
+#
+# The 'isolated_cores=' variable below controls which cores should be
+# isolated. By default we reserve 1 core per socket for housekeeping
+# and isolate the rest. But you can isolate any range as shown in the
+# examples below. Just remember to keep only one isolated_cores= line.
+#
+# Examples:
+# isolated_cores=2,4-7
+# isolated_cores=2-23
+#
+# Reserve 1 core per socket for housekeeping, isolate the rest.
+# Change this for a core list or range as shown above.
+isolated_cores=2-3 <1>
+
+#
+# Uncomment the 'isolate_managed_irq=Y' bellow if you want to move kernel
+# managed IRQs out of isolated cores. Note that this requires kernel
+# support. Please only specify this parameter if you are sure that the
+# kernel supports it.
+#
+isolate_managed_irq=Y <2>
 
+#
+# Set the desired combined queue count value using the parameter provided
+# below. Ideally this should be set to the number of housekeeping CPUs i.e.,
+# in the example given below it is assumed that the system has 4 housekeeping
+# (non-isolated) CPUs.
+#
+# netdev_queue_count=4
+----
+<1> The first two CPUs (0 and 1) are set aside for house keeping tasks and the rest are isolated for the real-time application.
+<2> isolate_managed_irq should be set to `Y`
+.. Activate the TuneD profile:
++
+[source,terminal]
+----
+$ tuned-adm profile realtime-virtual-guest
+----
+.. Set the real-time kernel as the default by using the GRUB boot loader command-line interface:
++
+[source,terminal]
+----
+$ grubby --set-default=<installed_rt_kernel_version>
```

Comment:
1. I'm using the U/S image `quay.io/kiagnose/kubevirt-realtime-checkup-vm:v0.2.0`, which produces a CentOS VM.
The kernel version as retrieved by either `uname -r` or `hostnamectl` is invalid for this command:
```
[cloud-user@realtime-vm tuned]$ uname -r
4.18.0-348.rt7.130.el8.x86_64
[cloud-user@realtime-vm tuned]$ sudo grubby --set-default=4.18.0-348.rt7.130.el8.x86_64
The param 4.18.0-348.rt7.130.el8.x86_64 is incorrect
```

[UPDATE] The cyclictest passed successfully (i.e. `Max Latencies` < 40us) despite this, so @orelmisan @RamLavi - is this setting necessary?

2. Also requires root

Reply:
`quay.io/kiagnose/kubevirt-realtime-checkup-vm:v0.2.0` is based on CentOS stream 8 and the real-time checkup does additional configuration to it after it boots.
Please test it with a "clean" RHEL 9 installation.